lib/state-store.js

```diff
@@ -859,26 +859,30 @@ class StateStore {
    * @param {string} checkpoint - The sequence number to store as the recovery point.
    * @param {string} shardsPath - The path pointing to where the shards state is stored.
    * @param {Object} shardsPathNames - The values of the attribute names in the path.
+   * @param {Date} timestamp - The approximate arrival timestamp
    * @fulfil {undefined}
    * @returns {Promise}
    */
-  async storeShardCheckpoint(shardId, checkpoint, shardsPath, shardsPathNames) {
+  async storeShardCheckpoint(shardId, checkpoint, shardsPath, shardsPathNames, timestamp) {
     if (typeof checkpoint !== 'string') throw new TypeError('The sequence number is required.');
+    if (typeof timestamp !== 'object' || timestamp === null) throw new TypeError('The timestamp is required.');
```

Comment:
timestamp is not required, if not provided we should not save it.
instead lets perform a check here
```js
let timestampString = null;
if (timestamp instanceof Date) {
  timestampString = timestamp.toISOString();
} else if (timestamp) {
  logger.warn("Will ignore timestamp provided, it was not a date: " + JSON.stringify(timestamp)); 
}
```